internal/pkg/agent/application/monitoring/server.go

```diff
@@ -61,8 +63,9 @@ func exposeMetricsEndpoint(
 
 	if enableProcessStats {
 		r.Handle("/processes", createHandler(processesHandler(coord)))
-		r.Handle("/processes/{processID}", createHandler(processHandler(coord, statsHandler)))
-		r.Handle("/processes/{processID}/", createHandler(processHandler(coord, statsHandler)))
+		r.Handle("/processes/{processID}", createHandler(processHandler(coord, statsHandler, operatingSystem)))
+		r.Handle("/processes/{processID}/", createHandler(processHandler(coord, statsHandler, operatingSystem)))
+		r.Handle("/processes/{processID}/{metricsPath}", createHandler(processHandler(coord, statsHandler, operatingSystem)))
```

Comment:
All these paths have the same handler?

Reply:
yes there is common special handling for agent in both of them (potentially) so we would end up with code dup either way. metricspath is a simple if check with a fallback to standard processing.